lib/stages/visRawReader.cpp

```diff
@@ -171,28 +172,34 @@ visRawReader::~visRawReader() {
     close(fd);
 }
 
-void visRawReader::change_dataset_state(dset_id_t ds_id) {
+void visRawReader::get_dataset_state(dset_id_t ds_id) {
     datasetManager& dm = datasetManager::instance();
 
-    // Add the states: metadata, time, prod, freq, input, eigenvalue and stack.
-    state_uptr sstate = nullptr;
-    if (!_stack.empty())
-        sstate = std::make_unique<stackState>(_num_stack, std::move(_rstack));
-    state_uptr istate = std::make_unique<inputState>(_inputs, std::move(sstate));
-    state_uptr evstate = std::make_unique<eigenvalueState>(_ev, std::move(istate));
-    state_uptr fstate = std::make_unique<freqState>(_freqs, std::move(evstate));
-    state_uptr pstate = std::make_unique<prodState>(_prods, std::move(fstate));
-    state_uptr tstate = std::make_unique<timeState>(_times, std::move(pstate));
-    state_uptr idstate = std::make_unique<acqDatasetIdState>(ds_id, std::move(tstate));
-
-    state_id_t mstate_id =
-        dm.add_state(std::make_unique<metadataState>(
-                         _metadata.at("weight_type"), _metadata.at("instrument_name"),
-                         _metadata.at("git_version_tag"), std::move(idstate)))
-            .first;
-
-    // register it as root dataset
-    _dataset_id = dm.add_dataset(mstate_id);
+    if (use_comet) {
+        // Add time to dataset
+        state_id_t tstate_id = dm.add_state(std::make_unique<timeState>(_times)).first;
+        // Register with dataset broker
+        _dataset_id = dm.add_dataset(tstate_id, ds_id);
+    } else {
```

Comment:
I guess my fear is somewhat that if we are going wholeheartedly down the `comet` route then this branch that intends it to allow gossec to work without comet is just going to atrophy if it's too different to the one we use.

Reply:
One way of merging them somewhat could be:
- Always register a `timeState` on top of the original dataset_id.
- If comet isn't available then you need to register all the other states too (inputState, prodState, ...), if comet is available you don't need to bother.
- Rewrite the dataset_id for each frame with all these new states in there.
- In `visTranspose` you simply look up the all the information you need from the current state, and then you write the output dataset_id as being the one that the `timeState` was registered against.

I think all put together that means there is a single branch here which adds the extra states when comet is not available, but everything else works fine.

Reply:
one problem is if comet is not available, you can't register a state on top of the original dataset_id because it doesn't exist as far as the dataset manager is concerned. So I think if we remove the branch that allows it to work without comet entirely we may need to drop that functionality entirely (this would be ok, I just left this in in case we wanted to transpose files manually for some reason)

Reply:
Right now if you run this without comet it ends up writing 0 to the `dataset_id` dataset in the HDF5 file because the base dataset is the root dataset. We could avoid this if it was possible to reconstruct a dataset state from a dataset ID but at the moment I don't know how you would do this.